### PR TITLE
Trace headers

### PR DESCRIPTION
**Pull request by derlih**
## What do these changes do?

Add a tracing signal to the client sent headers

## Are there changes in behavior for the user?

Yes

## Related issue number

#5105 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by derlih** [comment]
Currently there is one problem with the generated documentation. I don't see the link to the source code for class TraceRequestHeadersSentParams
Does anyone known how to fix it?

File: docs/tracing_reference.rst [hunk 1]
```diff
@@ -269,6 +269,13 @@ TraceConfig
 
       ``params`` is :class:`aiohttp.TraceDnsCacheMissParams` instance.
 
+   .. attribute:: on_request_headers_sent
+
+      Property that gives access to the signals that will be executed
+      when request headers are sent.
+
+      ``params`` is :class:`aiohttp.TraceRequestHeadersSentParams` instance.
+
```
Comment (asvetlov) [inline docs/tracing_reference.rst:278]: Please add `.. versionadded:: 3.8` here

File: docs/tracing_reference.rst [hunk 2]
```diff
@@ -492,3 +499,22 @@ TraceDnsCacheMissParams
    .. attribute:: host
 
        Host didn't find the cache.
+
+TraceRequestHeadersSentParams
+-----------------------------
+
+.. class:: TraceRequestHeadersSentParams
```
Comment (asvetlov) [inline docs/tracing_reference.rst:507]: `.. versionadded` for the class

**Review by asvetlov — COMMENTED** [review]
Looks good, please apply the last documentation polishing

File: docs/tracing_reference.rst (hunk 2 above)
Comment (derlih) [reply docs/tracing_reference.rst:507]: done

File: docs/tracing_reference.rst (hunk 1 above)
Comment (derlih) [reply docs/tracing_reference.rst:278]: done

**Comment by asvetlov** [comment]
FYI. We use squash-and-merge strategy for the project.
Thus, you *don't need *to rebase and force-push your changes.
A person who accepts your changes (me for example) is responsible to overview the commit message generated by GitHub and clean it up by removing unnecessary lines from in-progress commits.

Sure, but you still *can* use rebase and force-push if you want but it doesn't make much sense. The history becomes linear on PR merging, not on the stage of working on the PR.

File: docs/tracing_reference.rst [hunk 3]
```diff
@@ -492,3 +500,24 @@ TraceDnsCacheMissParams
...
+
+   .. versionadded:: 3.8
+
+   See :attr:`TraceConfig.on_request_headers_sent` for details.
```
Comment (asvetlov) [inline docs/tracing_reference.rst]:
```suggestion
   See :attr:`TraceConfig.on_request_headers_sent` for details.

   .. versionadded:: 3.8
```

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
LGTM, thanks!

**Pull request merged**